kani-driver/src/list/collect_metadata.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// This module invokes the compiler to gather the metadata for the list subcommand, then post-processes the output.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use crate::{
+    InvocationType,
+    args::list_args::{CargoListArgs, Format, StandaloneListArgs},
+    list::Totals,
+    list::output::{json, pretty},
+    project::{Project, cargo_project, standalone_project, std_project},
+    session::KaniSession,
+    version::print_kani_version,
+};
+use anyhow::Result;
+use kani_metadata::{ContractedFunction, HarnessKind, KaniMetadata};
+
+/// Process the KaniMetadata output from kani-compiler and output the list subcommand results
+fn process_metadata(metadata: Vec<KaniMetadata>, format: Format) -> Result<()> {
+    // We use ordered maps and sets so that the output is in lexographic order (and consistent across invocations).
+
+    // Map each file to a vector of its harnesses.
+    let mut standard_harnesses: BTreeMap<String, BTreeSet<String>> = BTreeMap::new();
+    let mut contract_harnesses: BTreeMap<String, BTreeSet<String>> = BTreeMap::new();
+
+    let mut contracted_functions: BTreeSet<ContractedFunction> = BTreeSet::new();
```

Comment:
I use ordered maps and sets because the data should be in alphabetical order. It would be weird if you ran list twice on the same inputs and your output was printed in a different order.

Reply:
See comment on line 21:

```rust
 // We use ordered maps and sets so that the output is in lexographic order (and consistent across invocations).
```